src/lib.rs

```diff
@@ -367,15 +370,23 @@ impl BedReader {
         // Get chunks that overlap this region
         let qchunks = self.index.query(tid, &region);
 
-        match (qchunks, self.reader) {
-            (Some(Ok(chunks)), BedReaderType::Bgzf(ref mut reader)) => {
+        // This is a dirty hack. Since noodles bgzf BufRead and Seek are only implemented on
+        // an owned Reader, by wrapping the reader in an Option, we can temporarily take ownership of
+        // it, and then put it back when we are done with it. Noodles should implement it's traits
+        // on both the owned and &mut refs of Reader.
+        let mut reader = self.reader.take().unwrap();
+
+        let ret = match (qchunks, reader) {
+            (Some(Ok(chunks)), BedReaderType::Bgzf(mut reader)) => {
                 let q = csi::io::Query::new(&mut reader, chunks);
                 let header = self.index.header().expect("No header found");
                 let q = q.indexed_records(&header).filter_by_region(&region);
                 Ok(q)
             }
             _ => unimplemented!(),
-        }
+        };
+        self.reader = Some(reader);
+        ret
```

Comment:
Note, the return type is now wrong, but that's expected I think since you have to convert out of the Noodes query return type to your own type.